### ISSUE TITLE
problem about tutorial_AC . why should 'v_' be placed here

### ISSUE THREAD

**Issue by 10ca1h0st** [tensorlayer/tensorlayer#1071]
Hello, i am newbie to rdl.
There is my question. Please help me.

The function is at examples/reinforcement_learning/tutorial_AC.py
def learn(self, state, reward, state_):
        v_ = self.model(np.array([state_]))
        with tf.GradientTape() as tape:
            v = self.model(np.array([state]))
            ## TD_error = r + lambda * V(newS) - V(S)
            td_error = reward + LAM * v_ - v
            loss = tf.square(td_error)
        grad = tape.gradient(loss, self.model.trainable_weights)
        self.optimizer.apply_gradients(zip(grad, self.model.trainable_weights))
        return td_error

My question is why the 'v_' is placed above  'with tf.GradientTape() as tape',
when I place 'v_ = self.model(np.array([state_]))'  in the with context, the model is not convergent.

Who can help me? Thanks. :)

**Comment by quantumiracle** [comment]
The 'v_' is target value, which is not supposed to be optimized. That's also why it's called the target, like the label in supervised learning.
